gqlopentelemetry/tracer.go

```diff
@@ -0,0 +1,132 @@
+package gqlopentelemetry
+
+import (
+	"context"
+	"fmt"
+	"go.opentelemetry.io/otel/api/global"
+	"go.opentelemetry.io/otel/codes"
+	"go.opentelemetry.io/otel/label"
+
+	"github.com/99designs/gqlgen/graphql"
+	"github.com/99designs/gqlgen/graphql/handler/extension"
+)
+
+type Tracer struct {
+	ComplexityExtensionName string
+	DataDog                 bool
+	Name                    string
```

Comment:
@k-yomo I was wrong about introducing `Name` field. I am sorry about that. I am going to test some changes locally and provide the feedback how we can use named tracer. Could you please delete `Name` for now?
```suggestion
```

Reply:
deleted!

Reply:
Looks like, tracer name is named by library not by user. 

https://github.com/open-telemetry/oteps/blob/master/text/0016-named-tracers.md#examples-of-tracer-and-meter-names

Reply:
Should it be `github.com/99designs/gqlgen-contrib`?

Reply:
@k-yomo Thanks for sharing!

Reply:
@k-yomo ie `const tracerName = "github.com/99designs/gqlgen-contrib/gqlopencensus"` and use it.

Reply:
There are no other places to use the tracer name, so I think we don't need to make it const.

Reply:
@k-yomo Could you please set to `""` everywhere?

`opentelemetry` will use default name provided by `TracerProvider`:

```json
{
    "InstrumentationLibrary": {
        "Name": "go.opentelemetry.io/otel/sdk/tracer",
        "Version": ""
    }
}
```

Reply:
What's the reason to use default name?

Reply:
@k-yomo You can override it when you configure `TracerProvider`.

Reply:
The user of this library can't override the name since there is  no way to manipulate.

Reply:
they set name in google's example.
https://github.com/GoogleCloudPlatform/opentelemetry-operations-go/blob/master/example/trace/http/client/client.go#L60

Setting name is not bad practice.

Reply:
@k-yomo That is true.

If you set it to `""` then `opentelemetry` will pick up the name from `TracerProvider`.

For now we can use `""` and refactor existing providers to use an approach where we can override `name`.

Do you agree?

Reply:
` refactor existing providers to use an approach where we can override name`
Sorry, I don't get especially the above part. Do you mean, we would be better to provide a way to override provider's 'name' to user?

Reply:
@k-yomo If we need to lock overriding tracer name, we also need to set it in two places. Please see my suggestions.

Reply:
@k-yomo I agree with you. Please see my suggestions.

Reply:
`There are no other places to use the tracer name`
oh sorry I missed second one, and ↑ is incorrect.

Reply:
Applied! Thank you for the suggestions!